### PR TITLE
Register AWSOSDMSP ClusterProfile with openshift ci

### PR DESCRIPTION
**Pull request by samvarankashyap**
$Subject
This is a initial pull request that starts work with registering awsosdmsp clusterprofile with openshift ci.

### CODE REVIEW

**Comment by samvarankashyap** [comment]
@bbguimaraes @droslean could you please take a look at it ?

File: cmd/repo-init/api.go [hunk 1]
```diff
@@ -718,6 +718,7 @@ func getClusterProfiles() []api.ClusterProfile {
 		api.ClusterProfileAzure2,
 		api.ClusterProfileAzure4,
 		api.ClusterProfileAzureStack,
+		api.ClusterProfileAWSOSDMSP,
```
[the remark this answers is not in the document]
Comment (samvarankashyap) [reply cmd/repo-init/api.go:721]: Sure @petr-muller  I will change it.

**Comment by samvarankashyap** [comment]
@petr-muller could you PTAL, Thanks.

File: cmd/repo-init/main.go [hunk 2]
```diff
@@ -817,6 +817,8 @@ func determineWorkflow(workflow string, clusterProfile api.ClusterProfile) *stri
 			ret = "ipi-alibabacloud"
 		case api.ClusterProfileNutanix:
 			ret = "ipi-nutanix"
+		case api.ClusterProfileAWSOSDMSP:
+			ret = "awsosdmsp"
```
Comment (petr-muller) [inline cmd/repo-init/main.go:821]: No such workflow exists: unless you plan to write it, I think it's fine to not add it into the `repo-init` tool.

File: pkg/api/types.go [hunk 3]
```diff
@@ -1092,6 +1092,7 @@ const (
 	ClusterProfileAzure4                ClusterProfile = "azure4"
 	ClusterProfileAzureArc              ClusterProfile = "azure-arc"
 	ClusterProfileAzureStack            ClusterProfile = "azurestack"
+	ClusterProfileAWSOSDMSP             ClusterProfile = "awsosdmsp"
```
Comment (petr-muller) [inline pkg/api/types.go:1095]: Sort this together with other AWS linkes, please. Also, please make this respect the convention and call it `aws-<name>` instead, like all other ones are called

File: pkg/api/types.go [hunk 4]
```diff
@@ -1153,6 +1154,7 @@ func ClusterProfiles() []ClusterProfile {
 		ClusterProfileAzure4,
 		ClusterProfileAzureArc,
 		ClusterProfileAzureStack,
+		ClusterProfileAWSOSDMSP,
```
Comment (petr-muller) [inline pkg/api/types.go:1157]: Sort with AWS

File: pkg/api/types.go [hunk 5]
```diff
@@ -1229,6 +1231,8 @@ func (p ClusterProfile) ClusterType() string {
 		return "azure4"
 	case ClusterProfileAzureStack:
 		return "azurestack"
+	case ClusterProfileAWSOSDMSP:
```
Comment (petr-muller) [inline pkg/api/types.go:1234]: Sort with AWS

File: pkg/api/types.go [hunk 6]
```diff
@@ -1330,6 +1334,8 @@ func (p ClusterProfile) LeaseType() string {
 		return "azure-arc-quota-slice"
 	case ClusterProfileAzureStack:
 		return "azurestack-quota-slice"
+	case ClusterProfileAWSOSDMSP:
+		return "awsosdmsp-quota-slice"
```
Comment (petr-muller) [inline pkg/api/types.go:1338]: Sort with AWS

File: pkg/api/types.go [hunk 7]
```diff
@@ -1412,7 +1418,7 @@ func (p ClusterProfile) LeaseType() string {
 // LeaseTypeFromClusterType maps cluster types to lease types
 func LeaseTypeFromClusterType(t string) (string, error) {
 	switch t {
-	case "aws", "aws-arm64", "aws-c2s", "aws-china", "aws-usgov", "aws-sc2s", "alibaba", "azure-2", "azure4", "azure-arc", "azurestack", "azuremag", "equinix-ocp-metal", "gcp", "libvirt-ppc64le", "libvirt-s390x", "nutanix", "openstack", "openstack-osuosl", "openstack-vexxhost", "openstack-ppc64le", "vsphere", "ovirt", "packet", "packet-edge", "kubevirt", "aws-cpaas", "osd-ephemeral":
+	case "aws", "aws-arm64", "aws-c2s", "aws-china", "aws-usgov", "aws-sc2s", "awsosdmsp", "alibaba", "azure-2", "azure4", "azure-arc", "azurestack", "azuremag", "equinix-ocp-metal", "gcp", "libvirt-ppc64le", "libvirt-s390x", "nutanix", "openstack", "openstack-osuosl", "openstack-vexxhost", "openstack-ppc64le", "vsphere", "ovirt", "packet", "packet-edge", "kubevirt", "aws-cpaas", "osd-ephemeral":
```
Comment (petr-muller) [inline pkg/api/types.go:1421]: `aws-<name>` convention

File: pkg/prowgen/podspec.go [hunk 8]
```diff
@@ -304,6 +304,7 @@ func generateClusterProfileVolume(profile cioperatorapi.ClusterProfile, clusterT
 		profile == cioperatorapi.ClusterProfilePacketSNO ||
 		profile == cioperatorapi.ClusterProfileAzureQE ||
 		profile == cioperatorapi.ClusterProfileAzureMagQE ||
+		profile == cioperatorapi.ClusterProfileAWSOSDMSP ||
```
Comment (petr-muller) [inline pkg/prowgen/podspec.go:307]: Sort with AWS

File: pkg/prowgen/podspec.go [hunk 9]
```diff
@@ -325,6 +326,7 @@ func generateClusterProfileVolume(profile cioperatorapi.ClusterProfile, clusterT
 		cioperatorapi.ClusterProfileAzure2,
 		cioperatorapi.ClusterProfileAzureArc,
 		cioperatorapi.ClusterProfileAzureStack,
+		cioperatorapi.ClusterProfileAWSOSDMSP,
```
Comment (petr-muller) [inline pkg/prowgen/podspec.go:329]: Sort with AWS

File: pkg/validation/test.go [hunk 10]
```diff
@@ -417,6 +417,7 @@ func validateClusterProfile(fieldRoot string, p api.ClusterProfile) []error {
 		api.ClusterProfileAzure4,
 		api.ClusterProfileAzureArc,
 		api.ClusterProfileAzureStack,
+		api.ClusterProfileAWSOSDMSP,
```
Comment (petr-muller) [inline pkg/validation/test.go:420]: Sort with AWS

**Comment by samvarankashyap** [comment]
@petr-muller I have updated the naming and also the order of the cluster profile . Please check.

**Comment by samvarankashyap** [comment]
@petr-muller @bbguimaraes @droslean could you PTAL 
Thank you.

**Pull request merged**